browser/src/control/Control.DocumentNameInput.js

```diff
@@ -60,7 +60,6 @@ window.L.Control.DocumentNameInput = window.L.Control.extend({
 			return;
 
 		$('#document-name-input').val(this.map['wopi'].BreadcrumbDocName);
-		this.map._onGotFocus();
 	},
```

Comment:
If I read this right, there are two places that are calling documentNameCancel.

One is the 'blur' handler, so I imagine what is happening is that when this widget loses focus then the focus is forced into the main document area, so it's impossible to keyboard navigate to the next widget?

I see another place, "onDocumentNameKeyPress" that calls "this.documentNameCancel" on pressing ESC. What happens before/after this change if you edit in that widget and press esc? Do we need to maybe put this.map._onGotFocus() after this.documentNameCancel() in that "if (e.keyCode === 27)" condition to keep that working as it used to?

Reply:
> One is the 'blur' handler, so I imagine what is happening is that when this widget loses focus then the focus is
 > forced into the main document area, so it's impossible to keyboard navigate to the next widget?
Yes

Reply:
> I see another place, "onDocumentNameKeyPress" that calls "this.documentNameCancel" on pressing ESC. What happens before/after this change if you edit in that widget and press esc? Do we need to maybe put this.map._onGotFocus() after this.documentNameCancel() in that "if (e.keyCode === 27)" condition to keep that working as it used to?

I haven’t tested it yet. By default, browsers only move focus using the Tab key and arrow keys inside focusable elements (like <input> or role="menuitem"). Other keys don’t change the focus.